reverse_engineering/api.js

```diff
@@ -21,19 +21,22 @@ const { prepareError } = require('./databaseService/helpers/errorService');
 
 module.exports = {
 	async connect(connectionInfo, logger, callback, app) {
-		const client = getClient();
-		const sshService = app.require('@hackolade/ssh-service');
+		const client = clientManager.getClient();
+
 		if (!client) {
-			await setClient(connectionInfo, sshService, 0, logger);
-			return getClient();
+			await clientManager.setClient({
+				connectionInfo,
+				logger,
+				sshService: app.require('@hackolade/ssh-service'),
```

Comment:
@chulanovskyi-bs looks like something that could follow `@hackolade/fetch` instead of depending back on the application at runtime

Reply:
(not for now obviously but we should think about it)

Reply:
sure, this was kept for backward compatibility, but agree, should be removed eventually